internal/resources/google/compute_external_vpn_gateway.go

```diff
@@ -0,0 +1,39 @@
+package google
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/infracost/infracost/internal/usage"
+)
+
+type ComputeExternalVPNGateway struct {
+	Address string
+	Region  string
+
+	MonthlyEgressDataTransferGB *ComputeExternalVPNGatewayNetworkEgressUsage `infracost_usage:"monthly_egress_data_transfer_gb"`
+}
+
+var ComputeExternalVPNGatewayUsageSchema = []*schema.UsageItem{
+	{
+		Key:          "monthly_egress_data_transfer_gb",
+		ValueType:    schema.SubResourceUsage,
+		DefaultValue: &usage.ResourceUsage{Name: "monthly_egress_data_transfer_gb", Items: ComputeExternalVPNGatewayNetworkEgressUsageSchema},
+	},
+}
+
+func (r *ComputeExternalVPNGateway) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *ComputeExternalVPNGateway) BuildResource() *schema.Resource {
+	region := r.Region
+	r.MonthlyEgressDataTransferGB.Region = region
```

Comment:
Do we need a check for this being nil, e.g.:

```go
if r.MonthlyEgressDataTransferGB == nil {
        r.MonthlyEgressDataTransferGB = &computeExternalVPNGatewayNetworkEgressUsage{}
}
```

Reply:
I think the call to `PopulateArgsWithUsage` will take care of initializing it.